### DISCUSSION TITLE
IRecoveryPolicy with Extensions.DependencyInjection

### DISCUSSION THREAD

**Discussion by Nimistren** [Havret/dotnet-activemq-artemis-client#442] [Q&A] [answered]
Hello, I am using ActiveMQ.Artemis.Client.Extensions.DependencyInjection as described here 
https://havret.io/activemq-artemis-net-core
I also found this link 
https://havret.github.io/dotnet-activemq-artemis-client/docs/connection-lifecycle, 
which describes how to get IRecoveryPolicy. But unfortunately I can't figure out how to correctly apply IRecoveryPolicy with DependencyInjection, because I don't see the AddActiveMq method with the ability to pass this policy

**Comment by Havret** [comment, marked as answer]
Hello @Nimistren,

You can set RecoveryPolicy via `ConfigureConnectionFactory` method:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddActiveMq(name: "my-artemis-cluster",
            endpoints: new[]
            {
                Endpoint.Create(host: "localhost", port: 5672, "artemis", "artemis"),
            })
        .ConfigureConnectionFactory((_, factory) =>
        {
            factory.RecoveryPolicy = RecoveryPolicyFactory.ExponentialBackoff(
                initialDelay: TimeSpan.FromSeconds(1),
                maxDelay: TimeSpan.FromSeconds(30),
                retryCount: 5);
        });
}
```

And here is the working example:
https://github.com/Havret/dotnet-activemq-artemis-client/blob/master/samples/ArtemisNetClient.Examples.AspNetCore/Startup.cs#L28

I hope that helps!
